R/lazy-join-query.R

```diff
@@ -190,11 +194,34 @@ sql_build.lazy_multi_join_query <- function(op, con, ..., sql_options = NULL) {
     }
   )
 
+  join_vars <- sql_multi_join_vars(
+    con,
+    op$vars,
+    table_vars,
+    use_star = FALSE,
+    qualify_all_columns = sql_options$qualify_all_columns
+  )
+  # WHERE happens after SELECT, but columns names are disambiguated using
+  # SELECT expressions, so need to backtransform
+  where <- lapply(op$where, \(expr) {
```

Comment:
Maybe refactor this into a function like `backtransform_where_table_vars()`

Reply:
Now think this PR is likely to be the right basic approach. I'll need to wrap up my work on dbplyr early in the new year and this PR does the job.